nifi-nar-bundles/nifi-evtx-bundle/nifi-evtx-processors/src/main/java/org/apache/nifi/processors/evtx/ParseEvtx.java

```diff
@@ -0,0 +1,352 @@
+@InputRequirement(Requirement.INPUT_REQUIRED)
+@Tags({"logs", "windows", "event", "evtx", "message", "file"})
+@CapabilityDescription("Parses the contents of a Windows Event Log file (evtx) and writes the resulting xml to the FlowFile")
+public class ParseEvtx extends AbstractProcessor {
+    public static final String RECORD = "Record";
+    public static final String CHUNK = "Chunk";
+    public static final String FILE = "File";
+    public static final String EVENTS = "Events";
+    public static final XMLOutputFactory XML_OUTPUT_FACTORY = XMLOutputFactory.newFactory();
+    public static final String EVTX_EXTENSION = ".evtx";
+    public static final String UNABLE_TO_PROCESS_DUE_TO = "Unable to process {} due to {}";
+    public static final String XML_EXTENSION = ".xml";
+
+    @VisibleForTesting
+    static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("Any FlowFile that was successfully converted from evtx to xml")
+            .build();
+
+    @VisibleForTesting
+    static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("Any FlowFile that encountered an exception during conversion will be transferred to this relationship with as much parsing as possible done")
+            .build();
+
+    @VisibleForTesting
+    static final Relationship REL_BAD_CHUNK = new Relationship.Builder()
+            .name("bad chunk")
+            .description("Any bad chunks of records will be transferred to this relationship in their original binary form")
+            .build();
+
+    @VisibleForTesting
+    static final Relationship REL_ORIGINAL = new Relationship.Builder()
+            .name("original")
+            .description("The unmodified input FlowFile will be transferred to this relationship")
+            .build();
+
+    @VisibleForTesting
+    static final Set<Relationship> RELATIONSHIPS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(REL_SUCCESS, REL_FAILURE, REL_ORIGINAL, REL_BAD_CHUNK)));
+
+    @VisibleForTesting
+    static final PropertyDescriptor GRANULARITY = new PropertyDescriptor.Builder().required(true)
+            .name("Granularity")
+            .description("Output flow file for each Record, Chunk, or File encountered in the event log")
+            .allowableValues(RECORD, CHUNK, FILE)
+            .build();
+
+    @VisibleForTesting
+    static final List<PropertyDescriptor> PROPERTY_DESCRIPTORS = Collections.unmodifiableList(Arrays.asList(GRANULARITY));
+
+    private final FileHeaderFactory fileHeaderFactory;
+    private final MalformedChunkHandler malformedChunkHandler;
+    private final RootNodeHandler rootNodeHandler;
+    private final XMLStreamWriterFactory xmlStreamWriterFactory;
+    private final ResultProcessor resultProcessor;
+
+    public ParseEvtx() {
+        this(FileHeader::new, ParseEvtx::handleMalformedChunkException, XmlBxmlNodeVisitor::new,
```

Comment:
You may want to stick to Java 7 or you will need to have a separate PR if we want to merge it in 0.x branch